### PR TITLE
add zap logger to controllers: pod, gwclass, gw, service

### PR DESCRIPTION
**Pull request by mikhail-aws**
Note:
- use zap logger vs wrapped version - zapr
- change "v" flag to "debug" to enable debug log
- use only 2 levels of logging info and debug
- replace glog with zap in following controllers: pod, service, gwclass, gateway
- partly address #126 

Logs example:
```
2023-08-09T20:56:50.126-0700	INFO	setup	aws-application-networking-k8s/main.go:97	init config	{"VpcId": "vpc-02aeb505678b5aad7", "Region": "us-west-2", "AccoundId": "566883718281", "DefaultServiceNetwork": "", "UseLongTgName": false}
...
2023-08-09T20:56:51.519-0700	DEBUG	datastore	latticestore/latticestore.go:169	AddServiceNetwork name:[my-hotel], account[566883718281] arn[arn:aws:vpc-lattice:us-west-2:566883718281:servicenetwork/sn-05bd7ca29c9532dcf], id[sn-05bd7ca29c9532dcf]
2023-08-09T20:56:51.519-0700	DEBUG	cloud	aws/cloud.go:26	request	{"serviceName": "VpcLattice", "operation": "ListServiceNetworks", "params": "{\n\n}"}
2023-08-09T20:56:51.589-0700	DEBUG	cloud	aws/cloud.go:38	response	{"serviceName": "VpcLattice", "operation": "ListServiceNetworks"}
2023-08-09T20:56:51.589-0700	INFO	controller.gateway	controllers/gateway_controller.go:208	successfully deployed model	{"stack": "my-hotel:default"}
2023-08-09T20:56:51.620-0700	DEBUG	controller.gateway	controllers/gateway_controller.go:238	serviceNetworkStatus : {{my-hotel 566883718281} arn:aws:vpc-lattice:us-west-2:566883718281:servicenetwork/sn-05bd7ca29c9532dcf sn-05bd7ca29c9532dcf service network is created and associated} for my-hotel
```

### CODE REVIEW

**Comment by rlymbur** [comment]
Thank you for tackling this. As a precaution, do E2E tests continue to pass?

**Comment by mikhail-aws** [comment]
> Thank you for tackling this. As a precaution, do E2E tests continue to pass?

I didnt run e2etests since only log statements are changed. But I ran a couple of manual testing steps.
```
make run
kubectl apply -f examples/my-hotel-gateway-tls.yaml
```

File: controllers/gateway_controller.go [hunk 1]
```diff
@@ -491,15 +474,13 @@ func UpdateGWListenerStatus(ctx context.Context, k8sclient client.Client, gw *ga
...
-		return errors.Wrapf(err, "failed to update gateway status")
+		return errors.Wrapf(err, "failed to update gateway listener err: %v, status: %v", err, gw.Status.Listeners)
 	}
 
 	if hasValidListener {
 		return nil
 	} else {
-		glog.V(2).Infof("no valid listeners for %v\n", gw.Name)
-		return errors.New("invalid listeners")
+		return fmt.Errorf("no valid listeners for %v\n", gw.Name)
```
Comment (rlymbur) [inline controllers/gateway_controller.go:483]: nit: Is a new line still required here?

File: controllers/service_controller.go [hunk 2]
```diff
@@ -48,32 +47,39 @@ const (
...
+	Client           client.Client
 	Scheme           *runtime.Scheme
 	finalizerManager k8s.FinalizerManager
 	eventRecorder    record.EventRecorder
 	modelBuilder     gateway.LatticeTargetsBuilder
 	stackDeployer    deploy.StackDeployer
 
-	latticeDataStore *latticestore.LatticeDataStore
-	stackMashaller   deploy.StackMarshaller
+	ds             *latticestore.LatticeDataStore
```
Comment (rlymbur) [inline controllers/service_controller.go:58]: nit: Can we align these?

File: pkg/latticestore/latticestore.go [hunk 3]
```diff
@@ -166,13 +169,13 @@ func (ds *LatticeDataStore) AddServiceNetwork(name string, account string, arn s
...
 
-	glog.V(6).Infof("AddServiceNetwork name:[%s], account[%s] arn[%s], id[%s]\n", name, account, arn, id)
+	ds.log.Debugf("AddServiceNetwork name:[%s], account[%s] arn[%s], id[%s]", name, account, arn, id)
 
 	Key := ServiceNetworkKey{Name: name, AccountID: account}
 	_, ok := ds.serviceNetworks[Key]
 
 	if ok {
-		glog.V(6).Infof("UpdateServiceNetwork name:[%s], account[%s] arn[%s] id[%s]\n", name, account, arn, id)
+		ds.log.Debugf("UpdateServiceNetwork name:[%s], account[%s] arn[%s] id[%s]\n", name, account, arn, id)
```
Comment (rlymbur) [inline pkg/latticestore/latticestore.go:178]: nit: I notice these have new lines as well. Are these still required? I remember you mentioning these may not be anymore.

File: cmd/aws-application-networking-k8s/main.go [hunk 4]
```diff
@@ -72,27 +72,37 @@ func main() {
...
+		setupLog.Error(err, "init config")
+		os.Exit(1)
+	}
+	setupLog.Infow("init config",
+		"VpcId", config.VpcID,
+		"Region", config.Region,
+		"AccoundId", config.AccountID,
+		"DefaultServiceNetwork", config.DefaultServiceNetwork,
+		"UseLongTgName", config.UseLongTGName,
+	)
```
Comment (xWink) [inline cmd/aws-application-networking-k8s/main.go:103]: Should we log every property individually here? My concern is that, as new init configs are added, this is one more place we need to keep updating. Can we just log all of the configs here (e.g. by logging the struct)?

File: controllers/gateway_controller.go [hunk 5]
```diff
@@ -163,43 +160,43 @@ func (r *GatewayReconciler) reconcile(ctx context.Context, req ctrl.Request) err
...
+				"gw", gw.Name+":"+gw.Namespace,
+			)
 			r.finalizerManager.RemoveFinalizers(ctx, gw, gatewayFinalizer)
 			return nil
 		}
 
 		return r.reconcileGatewayResources(ctx, gw)
 	} else {
-		gwLog.Info("Ignore non aws gateways!!!")
+		r.log.Infow("ignore non aws gateways!!!", "req", req)
```
Comment (xWink) [inline controllers/gateway_controller.go:185]: Do we need the !!! in here?

File: controllers/service_controller.go [hunk 6]
```diff
@@ -48,32 +47,39 @@ const (
...
-	client.Client
+	log              gwlog.Logger
+	Client           client.Client
 	Scheme           *runtime.Scheme
 	finalizerManager k8s.FinalizerManager
 	eventRecorder    record.EventRecorder
 	modelBuilder     gateway.LatticeTargetsBuilder
 	stackDeployer    deploy.StackDeployer
 
-	latticeDataStore *latticestore.LatticeDataStore
```
Comment (xWink) [inline controllers/service_controller.go:58]: Why make this less readable?

File: pkg/utils/gwlog/gwlog_test.go [hunk 7]
```diff
@@ -0,0 +1 @@
+package gwlog
```
Comment (xWink) [inline pkg/utils/gwlog/gwlog_test.go:1]: What's the purpose of this file?

**Review by xWink — COMMENTED** [review]
Love this move towards better logging. Thank you!

**Review by solmonk — COMMENTED** [review]
I know it is a too big of a PR to update and format every log lines, but at least let's make an announcement to get rid of `%v` and embed the structures whenever possible.

File: controllers/gateway_controller.go (hunk 1 above)
Comment (mikhail-aws) [reply controllers/gateway_controller.go:483]: no, missed \n, will fix other occurrences too

File: cmd/aws-application-networking-k8s/main.go (hunk 4 above)
Comment (mikhail-aws) [reply cmd/aws-application-networking-k8s/main.go:103]: Config is package name here, and all properties are package global variables. There is a ticket in the backlog to make them into a struct, but there are many places that refer to these, it's a noticeable refactoring. #310

File: Makefile [hunk 8]
```diff
@@ -45,7 +45,7 @@ help: ## Display this help.
 
 .PHONY: run
 run: ## Run in development mode
-	go run cmd/aws-application-networking-k8s/main.go
+	go run cmd/aws-application-networking-k8s/main.go --debug
```
[the remark this answers is not in the document]
Comment (mikhail-aws) [reply Makefile:48]: I'm thinking when you run controller in development mode you except it have debug logs.

File: pkg/utils/gwlog/gwlog_test.go (hunk 7 above)
Comment (mikhail-aws) [reply pkg/utils/gwlog/gwlog_test.go:1]: leftover after renaming log to gwlog, there was empty test file. will remove

File: controllers/service_controller.go (hunk 6 above)
Comment (mikhail-aws) [reply controllers/service_controller.go:58]: looks fine to me, but can return it back

File: controllers/gateway_controller.go (hunk 5 above)
Comment (mikhail-aws) [reply controllers/gateway_controller.go:185]: no idea!!!

**Comment by mikhail-aws** [comment]
addressed comments and ran full e2e suite
```
Ran 11 of 11 Specs in 4965.560 seconds
SUCCESS! -- 11 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestIntegration (4966.49s)
```

**Review by rlymbur — APPROVED** [review]
Looks good to me!

**Review by solmonk — APPROVED** [review]

**Pull request merged**